mozetl/clientsdaily/rollup.py

```diff
@@ -97,10 +183,23 @@ def write_by_activity_day(
         prefix = prefix_template.format(isoday)
         output_path = format_spark_path(output_bucket, prefix)
         data_for_date = results.where(results.activity_date == isoday)
-        data_for_date.coalesce(partition_count).write.parquet(output_path)
+        to_write = data_for_date.coalesce(partition_count)
+        to_write.write.parquet(output_path)
+        to_write.unpersist()
```

Comment:
Do you need to unpersist if you never persist the dataframe?

Reply:
Doc for unpersist(): "Marks the DataFrame as non-persistent, and remove all blocks for it from memory and disk." It's possible the implicit __del__ as the objects are unbound will be as fast, but maybe not. Also these lines emphasize how tight memory is on our clusters during this process.